File: finalAssignmentBackend/src/main/java/com/tutict/finalassignmentbackend/entity/ProgressItem.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("progress_items") // 指定表名
public class ProgressItem implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO) // 主键，自动递增
    private Integer id;

    @TableField("title") // 映射字段
    private String title;

    @TableField("status") // 映射字段
    private String status; // "Pending", "Processing", "Completed", "Archived"

    @TableField("submit_time")
    private LocalDateTime submitTime;

    @TableField("details") // 映射字段，允许为空
    private String details;

    @TableField("username") // 映射字段
    private String username; // 关联用户

    /**
     * 软删除时间戳
     */
    @TableField("deleted_at")
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;
}

File: final_asssignment_backend_go/project/internal/domain/progress_item.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// ProgressItem 表示 progress_items 表的实体
type ProgressItem struct {
	ID         int            `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	Title      string         `gorm:"column:title" json:"title"`
	Status     string         `gorm:"column:status" json:"status"` // "Pending", "Processing", "Completed", "Archived"
	SubmitTime time.Time      `gorm:"column:submit_time" json:"submit_time"`
	Details    string         `gorm:"column:details" json:"details"`
	Username   string         `gorm:"column:username" json:"username"`
	DeletedAt  gorm.DeletedAt `gorm:"index"` // 软删除字段（可选，GORM 默认支持）
}

// TableName 指定数据库表名
func (ProgressItem) TableName() string {
	return "progress_items"
}
